builtin/providers/aws/resource_aws_instance.go

```diff
@@ -174,37 +193,63 @@ func resourceAwsInstance() *schema.Resource {
 							ForceNew: true,
 						},
 
-						"delete_on_termination": &schema.Schema{
-							Type:     schema.TypeBool,
+						"volume_type": &schema.Schema{
+							Type:     schema.TypeString,
 							Optional: true,
-							Default:  true,
+							Computed: true,
 							ForceNew: true,
 						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%t-", m["delete_on_termination"].(bool)))
+					buf.WriteString(fmt.Sprintf("%s-", m["device_name"].(string)))
+					buf.WriteString(fmt.Sprintf("%t-", m["encrypted"].(bool)))
+					// NOTE: Not considering IOPS in hash; when using gp2, IOPS can come
+					// back set to something like "33", which throws off the set
+					// calculation and generates an unresolvable diff.
+					// buf.WriteString(fmt.Sprintf("%d-", m["iops"].(int)))
+					buf.WriteString(fmt.Sprintf("%s-", m["snapshot_id"].(string)))
+					buf.WriteString(fmt.Sprintf("%d-", m["volume_size"].(int)))
+					buf.WriteString(fmt.Sprintf("%s-", m["volume_type"].(string)))
+					return hashcode.String(buf.String())
```

Comment:
As part of this adventure I learned that `TypeSet` attributes need to change their hash code if they want any sub-attribute to be detected as a change, because if the hashcodes match, the contents of the Set are assumed to be the same.